hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -247,24 +248,26 @@ private void interruptThread(Thread thread) {
         }
     }
 
-    public Object callOnAddress(Address address, Operation operation) {
+    public InternalCompletableFuture<Object> callOnAddress(Address address, Operation operation) {
         // TODO: why are we always executing on the MapService?
         OperationService operationService = instance.node.nodeEngine.getOperationService();
-        Future future = operationService.invokeOnTarget(MapService.SERVICE_NAME, operation, address);
-        try {
-            return future.get();
-        } catch (Throwable t) {
-            return ExceptionUtil.toString(t);
-        }
+        return operationService.invokeOnTarget(MapService.SERVICE_NAME, operation, address);
     }
 
-    public Object callOnThis(Operation operation) {
+    public InternalCompletableFuture<Object> callOnThis(Operation operation) {
         return callOnAddress(instance.node.getThisAddress(), operation);
     }
 
-    public Object callOnMember(Member member, Operation operation) {
-        Address address = member.getAddress();
-        return callOnAddress(address, operation);
+    public InternalCompletableFuture<Object> callOnMember(Member member, Operation operation) {
+        return callOnAddress(member.getAddress(), operation);
+    }
+
+    public static Object resolveFuture(Future<Object> future) {
+        try {
+            return future.get();
```

Comment:
This is not right, but I didn't want to go ahead and fix this for all requests from Management Center as it is out of scope for this PR which is enabling MC to distinguish between an error condition vs a successful operation for the operations that we print the error stacktrace on the UI. These operations are `ThreadDumpRequest` and `ExecuteScriptRequest`.